cleanlab/datalab/internal/issue_manager/underperf_group.py

```diff
@@ -126,24 +127,84 @@ def set_knn_graph(self, features: npt.NDArray, find_issues_kwargs: Dict) -> csr_
             except NotFittedError:
                 knn.fit(features)
 
-            knn_graph = knn.kneighbors_graph(mode="distance", n_neighbors=features.shape[0] - 1)
-        return knn_graph
+            knn_graph = knn.kneighbors_graph(mode="distance")
+            symmetric_knn_graph = self._symmetrize_knn_graph(knn_graph)
+        return symmetric_knn_graph, knn_graph
+
+    def _symmetrize_knn_graph(self, knn_graph: csr_matrix) -> csr_matrix:
+        binary_adjacency = (knn_graph > 0).astype(int)
+        symmetric_adjacency = binary_adjacency.maximum(binary_adjacency.T)
+        symmetric_distances = knn_graph + knn_graph.T.multiply(
+            symmetric_adjacency - binary_adjacency
+        )
+        return symmetric_distances
+
+    def perform_clustering(self, knn_graph: csr_matrix) -> npt.NDArray[np.int_]:
+        # hdb = HDBSCAN(metric="precomputed")
+        # hdb.fit(knn_graph)
+        # cluster_labels = hdb.labels_
+        # ALTERNATIVE FORMULATION: Connect the graph using a dummy node before HDBSCAN.
+        # Adapted from - https://github.com/scikit-learn-contrib/hdbscan/issues/82#issuecomment-1242604639
+        # knn_graph = self.connect_knn_graph(knn_graph)
+        cluster_labels = self.run_hdbscan_on_components(knn_graph)
```

Comment:
Could we consider using a different variable name than `knn_graph` when transformations such as symmetrizing or adding any kind of bridge edges are anticipated?

Additionally, would it be possible to set [DBSCAN](https://scikit-learn.org/stable/modules/generated/sklearn.cluster.DBSCAN.html#sklearn-cluster-dbscan) as the default clustering algorithm for the time being? This would ensure we have an algorithm that is guaranteed to function with an existing knn graph without necessitating excessive computation.

Note that it should operate with the knn_graph once it has been made symmetric, but it shouldn’t require a connected knn graph.